core/list/list.c

```diff
@@ -0,0 +1,68 @@
+#include "list.h"
+
+#include <stdlib.h>
+
+struct List *empty_list() {
+  struct List *list_ptr = (struct List *)malloc(sizeof(struct List));
+  struct List list_v = {
+      .head = NULL,
+      .tail = NULL,
+  };
+  *list_ptr = list_v;
+  return list_ptr;
+}
+
+struct List *push(struct List *list, void *element) {
+  list->tail = empty_list();
+  list->head = element;
+  return list->tail;
+}
+
+struct List *new_list(int size, void **values) {
+  struct List *head = empty_list();
+  struct List *next = head;
+  int index = 0;
+  while (index < size) {
+    next = push(next, *(values + index));
+    index++;
+  }
+  return head;
+}
+
+uint64_t list_length(struct List *list) {
```

Comment:
Why don't you maintain a pointer of the length variable, and needs to loop the list?

Reply:
Or it's an original package provided by C?

Reply:
yes

Reply:
Basically, manage the length cache, for now, is too complicated, so just need to sure functionality is correct.